core/src/main/java/org/elasticsearch/action/admin/indices/analyze/AnalyzeRequest.java

```diff
@@ -82,35 +136,77 @@ public String analyzer() {
     }
 
     public AnalyzeRequest tokenizer(String tokenizer) {
-        this.tokenizer = tokenizer;
+        if (tokenizer != null) {
+            XContentType type = XContentFactory.xContentType(tokenizer);
+            if (type == null) {
+                this.tokenizer = new NameOrDefinition(tokenizer);
+            } else {
+                try {
+                    Settings settings = Settings.builder().loadFromSource(tokenizer).build();
+                    this.tokenizer = new NameOrDefinition(settings);
+                } catch (Exception e) {
+                    throw new IllegalArgumentException("failed to parse tokenizer[" + tokenizer + "]", e);
+                }
+            }
+        } else {
+            this.tokenizer = null;
+        }
         return this;
     }
 
-    public String tokenizer() {
+    public AnalyzeRequest tokenizer(Map<String, Object> tokenizer) {
+        this.tokenizer = new NameOrDefinition(tokenizer);
+        return this;
+    }
+
+    public NameOrDefinition tokenizer() {
         return this.tokenizer;
     }
 
-    public AnalyzeRequest tokenFilters(String... tokenFilters) {
-        if (tokenFilters == null) {
-            throw new IllegalArgumentException("token filters must not be null");
+    public AnalyzeRequest addTokenFilter(String tokenFilter) {
+        XContentType type = XContentFactory.xContentType(tokenFilter);
+        if (type == null) {
+            this.tokenFilters.add(new NameOrDefinition(tokenFilter));
+        } else {
+            try {
+                Settings settings = Settings.builder().loadFromSource(tokenFilter).build();
+                this.tokenFilters.add(new NameOrDefinition(settings));
+            } catch (Exception e) {
+                throw new IllegalArgumentException("failed to parse token filter[" + tokenFilter + "]", e);
+            }
         }
-        this.tokenFilters = tokenFilters;
         return this;
     }
 
-    public String[] tokenFilters() {
+    public AnalyzeRequest addTokenFilter(NameOrDefinition tokenFilter) {
```

Comment:
Since we already have `addTokenFilter(String)`, this method is only useful to provide a definition, so maybe it should be `addTokenFilter(Map<String, ?> tokenFilter)`?